payment/key_test.go

```diff
@@ -2,22 +2,36 @@ package payment
 
 import (
 	"encoding/base64"
+	"encoding/hex"
+	"fmt"
 	"testing"
 	"time"
+
+	cryptography "github.com/brave-intl/bat-go/utils/cryptography"
 )
 
 func TestGenerateSecret(t *testing.T) {
 	// set up the aes key, typically done with env variable atm
-	oldAESKey := AESKey
+	oldEncryptionKey := cryptography.EncryptionKey
 	defer func() {
-		AESKey = oldAESKey
+		cryptography.EncryptionKey = oldEncryptionKey
 	}()
-	AESKey = "123456789012345678901234"
+	cryptography.EncryptionKey = []byte("MTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0")
 	s, n, err := GenerateSecret()
 	if err != nil {
 		t.Error("error in generate secret: ", err)
 	}
-	secretKey, err := decryptSecretKey(s, n)
+
+	encrypted, err := hex.DecodeString(s)
+	if err != nil {
+		t.Error("error while decoding the encrypted string", err)
+	}
+	nonce, err := hex.DecodeString(n)
+	if err != nil {
+		t.Error("error while decoding the nonce", err)
+	}
+
+	secretKey, err := cryptography.DecryptMessage(encrypted, nonce)
 	if err != nil {
 		t.Error("error in decrypt secret: ", err)
 	}
```

Comment:
maybe add a test below that nonce and secretKey (and anything else that is fixed length) has the expected length

Reply:
Added! https://github.com/brave-intl/bat-go/pull/347/commits/6396ea948031c9eb0580a66442e3e3342bb731de